spring-cloud-commons/src/main/java/org/springframework/cloud/commons/publisher/FluxFirstNonEmptyEmitting.java

```diff
@@ -277,23 +293,26 @@ int resignFromRace() {
 		@Nullable
 		public Object scanUnsafe(Attr key) {
 			if (key == Attr.PARENT) {
-				return s;
+				return ReflectionUtils.getField(s, this);
```

Comment:
this should actually work without reflection: resolve class-specific state first, then delegate to `super` for the rest. `DeferredSubscription` already answers `s` for the `Attr.PARENT` key.